src/main/java/core/basesyntax/FieldXY.java

```diff
@@ -1,19 +1,31 @@
 package core.basesyntax;
 
-/**
- * <p>На игровом поле находится робот. Позиция робота на поле описывается двумя
- * целочисленным координатами: X и Y. 
- * Ось X смотрит слева направо, ось Y — снизу вверх.(Помните, как рисовали 
- * графики функций в школе?)
- * В начальный момент робот находится в некоторой позиции на поле.
- * Также известно, куда робот смотрит: вверх, вниз, направо или налево.
- * Ваша задача — привести робота в заданную точку игрового поля.
- * Реализуйте метод moveRobot()</p>
- */
 public class FieldXY {
 
     public void moveRobot(Robot robot, int toX, int toY) {
-        //your code
+        if (toX > robot.getX()) {
+            while (robot.getDirection() != Direction.RIGHT) {
+                robot.turnRight();
+            }
+        } else {
+            while (robot.getDirection() != Direction.LEFT) {
+                robot.turnLeft();
+            }
+        }
+        while (robot.getX() != toX) {
+            robot.stepForward();
+        }
+        if (toY > robot.getY()) {
+            while (robot.getDirection() != Direction.UP) {
+                robot.turnRight();
+            }
+        } else {
+            while (robot.getDirection() != Direction.DOWN) {
+                robot.turnRight();
+            }
+        }
+        while (robot.getY() != toY) {
+            robot.stepForward();
+        }
```

Comment:
Try to divide your code by logic and create methods out of it. It's better to have small blocks of code.